doc/extending/extending_theano_gpu.txt

```diff
@@ -78,3 +78,57 @@ If you don't have any input variables on the GPU you can follow the
 the example of :class:`theano.gpuarray.basic_ops.GpuFromHost` or
 :class:`theano.gpuarray.basic_ops.GpuEye`.  This is not a case that
 you should encounter often, so it will not be covered further.
+
+Defining new kernels
+====================
+
+If your op needs to do some transformation on the data, chances are
+that you will need to write a new kernel.  The best way to do this is
+to leverage GpuKernelBase (or CGpuKernelBase if you want to use the
+COp functionality).
+
+For plain GpuKernelBase, you have to define a method called
+gpu_kernels which returns a list of :class:`Kernel
+<theano.gpuarray.basic_ops.Kernel>` objects.  You can define as many
+kernels as you want for a single op.  An example would look like this:
+
+    def gpu_kernels(self, node, name):
+        code = """
+KERNEL void k(GLOBAL_MEM ga_float *a, ga_size n, ga_size m) {
+    ga_size nb = n < m ? n : m;
+    for (ga_size i = LID_0; i < nb; i += LDIM_0) {
+        a[i*m + i] = 1;
+    }
+}"""
+        return [Kernel(
+                code=code, name="k",
+                params=[gpuarray.GpuArray, gpuarray.SIZE, gpuarray.SIZE],
+                flags=Kernel.get_flags(self.dtype))]
+
+If you want to use COp, then you should use `CGpuKernelBase` instead.
```

Comment:
Make a full example with CGpuKernelBase and put this as a test and load it in the doc. To make sure the doc is up to date and that we don't break it.